### PR TITLE
feat: allow callers to specify custom IAM path for instance profiles

### PR DESCRIPTION
**Pull request by DerekFrank**
Fixes #N/A 

**Description**
- Changes `instanceprofile.Provider.Create()` to accept a `path string` parameter instead of `usePath bool`
- When `path` is non-empty, it is set directly on the `CreateInstanceProfile` input, allowing downstream consumers to specify custom IAM path prefixes
- The upstream karpenter-provider-aws caller now explicitly passes the `/karpenter/{region}/{clusterName}/{nodeClassUID}/` path string

**How was this change tested?**

`make presubmit`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/providers/instanceprofile/instanceprofile.go [hunk 1]
```diff
@@ -93,7 +102,7 @@ func (p *DefaultProvider) Create(
 	roleName string,
 	tags map[string]string,
 	nodeClassUID string,
```
Comment (jigisha620) [inline pkg/providers/instanceprofile/instanceprofile.go]: `nodeClassUID` is not needed anymore now that we are using `path` right?

File: pkg/providers/instanceprofile/suite_test.go [hunk 2]
```diff
@@ -109,12 +109,15 @@ var _ = AfterEach(func() {
...
+	karpenterPath := func(nodeClassUID string) string {
+		return instanceprofile.FormatPath("karpenter", fake.DefaultRegion, options.FromContext(ctx).ClusterName, nodeClassUID)
+	}
 	DescribeTable(
 		"should support IAM roles",
 		func(roleWithPath, role string) {
 			const profileName = "profile-A"
 			nodeClass.Spec.Role = roleWithPath
-			Expect(awsEnv.InstanceProfileProvider.Create(ctx, profileName, role, nil, string(nodeClass.UID), true)).To(Succeed())
+			Expect(awsEnv.InstanceProfileProvider.Create(ctx, profileName, role, nil, string(nodeClass.UID), karpenterPath(string(nodeClass.UID)))).To(Succeed())
```
Comment (jigisha620) [inline pkg/providers/instanceprofile/suite_test.go]: Can we also add a test that verifies the flow for empty path?

**Review by jigisha620 — COMMENTED** [review]
`ListClusterProfiles` and `ListNodeClassProfiles` still use the old path so if the downstream consumers use a different path then the garbage collector will not be able to find orphaned profiles.

File: pkg/providers/instanceprofile/instanceprofile.go (hunk 1 above)
Comment (DerekFrank) [reply pkg/providers/instanceprofile/instanceprofile.go]: Good point!

**Review by jigisha620 — APPROVED** [review]
LGTM 🚀
